src/app/Library/CrudPanel/CrudColumn.php

```diff
@@ -38,15 +38,15 @@ class CrudColumn
 
     protected $attributes;
 
-    public function upload($upload = true)
+    public function __construct($nameOrDefinitionArray)
     {
-        $this->attributes['upload'] = $upload;
-
-        return $this->save();
-    }
+        if (is_array($nameOrDefinitionArray)) {
+            $column = $this->crud()->addAndReturnColumn($nameOrDefinitionArray);
+            $name = $column->getAttributes()['name'];
```

Comment:
Why create a new method when we already have one that does that?

```suggestion
            $this->crud()->addAndReturnColumn($nameOrDefinitionArray);
            $name = $nameOrDefinitionArray['name'];
```

Reply:
What if there is no name? 
Columns don't have a mandatory name. 

That's why you need the created column to get the anonymous name.